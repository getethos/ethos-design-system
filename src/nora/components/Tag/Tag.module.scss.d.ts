// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'Cyan': string;
  'Default': string;
  'Green': string;
  'Neutral': string;
  'Orange': string;
  'Red': string;
  'Tag': string;
}
export const cssExports: CssExports;
export default cssExports;
