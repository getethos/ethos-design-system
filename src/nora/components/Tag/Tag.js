import React from 'react'
import PropTypes from 'prop-types'
import styles from './Tag.module.scss'

export const Tag = ({ children, type }) => {
  let colorClass
  switch (type) {
    case 'red':
      colorClass = styles.Red
      break
    case 'orange':
      colorClass = styles.Orange
      break
    case 'green':
      colorClass = styles.Green
      break
    case 'cyan':
      colorClass = styles.Cyan
      break
    default:
      colorClass = styles.Default
  }

  return <div className={[styles.Tag, colorClass].join(' ')}>{children}</div>
}

Tag.propTypes = {
  type: PropTypes.oneOf(['red', 'orange', 'green', 'cyan']),
  children: PropTypes.string.isRequired,
}

Tag.defaultProps = {
  type: undefined,
}
