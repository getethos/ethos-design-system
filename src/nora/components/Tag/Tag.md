### Text info messages

Default tag
```jsx
<Tag>Preview Text</Tag>
```

Rejected tag
```jsx
<Tag type='green'>Preview Text</Tag>
```

