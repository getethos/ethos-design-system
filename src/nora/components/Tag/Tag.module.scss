@import '../Attributes.css';

.Tag {
  border-radius: var(--Radius-3);
  display: inline-block;
  align-items: center;
  padding: var(--Space-4);
  font-weight: 500;
}

.Red {
  background-color: var(--NoraRed-100);
  color: var(--NoraRed-400);
}

.Orange {
  background-color: var(--NoraOrange-100);
  color: var(--NoraOrange-400);
}

.Green {
  background-color: var(--NoraGreen-100);
  color: var(--NoraGreen-400);
}

.Cyan {
  background-color: var(--NoraCyan-100);
  color: var(--NoraCyan-400);
}

.Default {
  background-color: var(--NoraSilver-100);
  color: var(--NoraSilver-400);
}
