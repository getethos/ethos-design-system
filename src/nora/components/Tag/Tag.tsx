import * as React from 'react'
import styles from './Tag.module.scss'
type TagProps = {
  type?: 'red' | 'orange' | 'green' | 'cyan' | 'gray' | 'neutral'
}
export const Tag: React.FC<TagProps> = ({ children, type }) => {
  let colorClass
  switch (type) {
    case 'red':
      colorClass = styles.Red
      break
    case 'orange':
      colorClass = styles.Orange
      break
    case 'green':
      colorClass = styles.Green
      break
    case 'cyan':
      colorClass = styles.Cyan
      break
    case 'neutral':
      colorClass = styles.Neutral
      break
    case 'gray':
      colorClass = styles.Default
      break
    default:
      colorClass = styles.Default
  }
  return <div className={[styles.Tag, colorClass].join(' ')}>{children}</div>
}
Tag.defaultProps = {
  type: undefined,
}
