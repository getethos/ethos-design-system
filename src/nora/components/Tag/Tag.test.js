import React from 'react'
import renderer from 'react-test-renderer'
import { Tag } from './Tag'

describe('Tag', () => {
  it('renders default tag', () => {
    const tree = renderer.create(<Tag>yo</Tag>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('red', () => {
    const tree = renderer.create(<Tag type="red">red</Tag>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('green', () => {
    const tree = renderer.create(<Tag type="green">green</Tag>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('orange', () => {
    const tree = renderer.create(<Tag type="orange">orange</Tag>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('cyan', () => {
    const tree = renderer.create(<Tag type="cyan">cyan</Tag>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
