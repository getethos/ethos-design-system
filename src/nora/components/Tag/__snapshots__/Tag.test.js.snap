// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag cyan 1`] = `
<div
  className="Tag Cyan"
>
  cyan
</div>
`;

exports[`Tag gray 1`] = `
<div
  className="Tag Default"
>
  gray
</div>
`;

exports[`Tag green 1`] = `
<div
  className="Tag Green"
>
  green
</div>
`;

exports[`Tag orange 1`] = `
<div
  className="Tag Orange"
>
  orange
</div>
`;

exports[`Tag red 1`] = `
<div
  className="Tag Red"
>
  red
</div>
`;

exports[`Tag renders default tag 1`] = `
<div
  className="Tag Default"
>
  yo
</div>
`;
